rustls/src/client/tls13.rs

```diff
@@ -269,6 +269,7 @@ pub(super) fn prepare_resumption(
     exts: &mut Vec<ClientExtension>,
     doing_retry: bool,
 ) {
+    cx.common.suite = Some(resuming_suite.into());
```

Comment:
The only readers of the `CommonState::suite` field other than `Connection::negotiated_cipher_suite()` are `quic::write_hs()` and `quic::next_1rtt_keys()`, both of which will only read it after handshake secrets have been established (in other words, on the client side the "new" suite will be available), as well as `<ServerConnection as QuicExt>::zero_rtt_keys()`, which is irrelevant by virtue of being server-side only. (With #750, `write_hs()` and `next_1rtt_keys()` will no longer rely on `CommonState::suite` either, and will get it directly from the handshake state.)